pkg/kapis/oauth/register.go

```diff
@@ -50,42 +53,65 @@ func AddToContainer(c *restful.Container, im im.IdentityManagementInterface,
+		Param(ws.QueryParameter("client_id", "OAuth 2.0 Client Identifier valid at the Authorization Server.").Required(true)).
+		Param(ws.QueryParameter("redirect_uri", "Redirection URI to which the response will be sent. "+
+			"This URI MUST exactly match one of the Redirection URI values for the Client pre-registered at the OpenID Provider.").Required(true)).
+		Param(ws.QueryParameter("scope", "OpenID Connect requests MUST contain the openid scope value. "+
+			"If the openid scope value is not present, the behavior is entirely unspecified.").Required(false)).
+		Param(ws.QueryParameter("state", "Opaque value used to maintain state between the request and the callback.").Required(false)).
+		To(handler.authorize).
+		Metadata(restfulspec.KeyOpenAPITags, []string{constants.AuthenticationTag}))
+
+	// Authorization Servers MUST support the use of the HTTP GET and POST methods
+	// defined in RFC 2616 [RFC2616] at the Authorization Endpoint.
+	ws.Route(ws.POST("/authorize").
+		Consumes(contentTypeFormData).
+		Doc("The authorization endpoint is used to interact with the resource owner and obtain an authorization grant.").
+		Param(ws.BodyParameter("response_type", "The value MUST be one of \"code\" for requesting an "+
+			"authorization code as described by [RFC6749] Section 4.1.1, \"token\" for requesting an access token (implicit grant)"+
+			" as described by [RFC6749] Section 4.2.2.").Required(true)).
+		Param(ws.BodyParameter("client_id", "OAuth 2.0 Client Identifier valid at the Authorization Server.").Required(true)).
+		Param(ws.BodyParameter("redirect_uri", "Redirection URI to which the response will be sent. "+
+			"This URI MUST exactly match one of the Redirection URI values for the Client pre-registered at the OpenID Provider.").Required(true)).
+		Param(ws.BodyParameter("scope", "OpenID Connect requests MUST contain the openid scope value. "+
+			"If the openid scope value is not present, the behavior is entirely unspecified.").Required(false)).
+		Param(ws.BodyParameter("state", "Opaque value used to maintain state between the request and the callback.").Required(false)).
+		To(handler.authorize).
 		Metadata(restfulspec.KeyOpenAPITags, []string{constants.AuthenticationTag}))
-	// Resource Owner Password Credentials Grant
+
 	// https://tools.ietf.org/html/rfc6749#section-4.3
 	ws.Route(ws.POST("/token").
-		Consumes("application/x-www-form-urlencoded").
+		Consumes(contentTypeFormData).
 		Doc("The resource owner password credentials grant type is suitable in\n"+
 			"cases where the resource owner has a trust relationship with the\n"+
 			"client, such as the device operating system or a highly privileged application.").
 		Param(ws.FormParameter("grant_type", "Value MUST be set to \"password\".").Required(true)).
 		Param(ws.FormParameter("username", "The resource owner username.").Required(true)).
 		Param(ws.FormParameter("password", "The resource owner password.").Required(true)).
-		To(handler.Token).
+		To(handler.token).
```

Comment:
Please add all the required `FormParameter`.

Reply:
done